File: tests/XMP/global-view/bcast/C/bcast3.c
#include <xmp.h>
#include <stdio.h>
#pragma xmp nodes p[*]

int hoge(int n1, int n2)
{
  int a[n1][n2];

  for(int i=0;i<10;i++)
    for(int j=0;j<10;j++)
      a[i][j] = 0;

  for(int i=0;i<10;i++)
    for(int j=0;j<10;j++)
      a[i][j] = xmp_node_num();
      
#pragma xmp bcast (a)

  int sum = 0;
  for(int i=0;i<10;i++)
    for(int j=0;j<10;j++)
      sum += a[i][j];

  int flag = (sum == 100)? 0 : 1;
#pragma xmp reduction(max:flag)
  if(xmp_node_num() == 1){
    if(flag == 0){
      printf("PASS\n");
    }
    else{
      printf("Error!\n");
      xmp_exit(1);
    }
  }
}

int main()
{
  hoge(10, 10);

  return 0;
}

File: tests/XMP/global-view/bcast/F/bcast3.f90
program main
  include 'xmp_lib.h'
  integer,parameter:: N=1000
  !$xmp nodes p(*)
  integer aa(N), a
  real*4 bb(N), b
  real*8 cc(N), c
  integer procs, result
  
  procs = xmp_num_nodes()
  result = 0
  do j=1, procs
     a = xmp_node_num()
     b = real(a)
     c = dble(a)
     do i=1, N
        aa(i) = a+i-1
        bb(i) = real(a+i-1)
        cc(i) = dble(a+i-1)
     enddo
     
     !$xmp bcast (a) from p(j)
     !$xmp bcast (b) from p(j)
     !$xmp bcast (c) from p(j)
     !$xmp bcast (aa) from p(j)
     !$xmp bcast (bb) from p(j)
     !$xmp bcast (cc) from p(j)

     if(a .ne. j) result = -1
     if(b .ne. real(j)) result = -1
     if(c .ne. dble(j)) result = -1
     do i=1, N
        if(aa(i) .ne. j+i-1) result = -1
        if(bb(i) .ne. real(j+i-1)) result = -1
        if(cc(i) .ne. dble(j+i-1)) result = -1
     enddo
  enddo

  !$xmp reduction(+:result)
  !$xmp task on p(1)
  if( result .eq. 0 ) then
     write(*,*) "PASS"
  else
     write(*,*) "ERROR"
     call exit(1)
  endif
  !$xmp end task
  
end program main
